docs/ObserveApiProposal.md

```diff
@@ -0,0 +1,782 @@
+completion.  Subscriptions would be offered to stay informed of progress.  It is expected that this
+would be at a higher level than the state maintained by Observe itself, but would provide a
+centralized, shared concept of the current state of execution to all interested parties.  Instead
+of "Observe tells Engage" that an acquisition is complete for example, what would actually 
+happen in the implementation would be that Engage monitors the observation execution state.
+
+Since it won't be needed by Observe, at least initially, we'll leave this for the future.
+
+
+### `TimeAccounting`
+
+`TimeAccounting` calculates actual executed time and categorizes it into program, partner, and
+non-charged categories. Detailed step-by-step, dataset-by-dataset time data can be found in the
+execution events discussed later.  Time accounting calculations will require observation events 
+along with weather and fault information.  Most of this is of little interest to Observe, though it
+will want to update the progress bar on individual steps and on the observation as a whole.
+
+_The following SDL sketches out the general idea but specific details are likely to change._
+
+```
+# Equivalent time amount in several unit options (e.g., 120 seconds or 2 minutes)
+type Duration {
+  # Duration in µs
+  microseconds: Long!
+
+  # Duration in ms
+  milliseconds: BigDecimal!
+
+  # Duration in seconds
+  seconds: BigDecimal!
+
+  # Duration in minutes
+  minutes: BigDecimal!
+
+  # Duration in hours
+  hours: BigDecimal!
+}
+
+# Summary of the time spent on an observation and how it is divided
+type TimeAccounting {
+
+   # Total execution time including program, partner and non-charged times
+   total:       Duration!
+
+   # Time subtracted from the program's time budget.
+   programTime: Duration!
+
+   # Time associated with the program's partners.
+   partnerTime: Duration!
+
+   # Time that is not charged to the program or partners (e.g., spent on
+   # datasets that do not meet quality standards.)
+   nonCharged:  Duration!
+
+}
+```
+
+### `PlannedTime`
+
+Remaining time, as opposed to `TimeAccounting`, deals with an estimation of _future_ execution time.
+Because the corresponding steps have not been executed, this is subject to change over time but 
+important for the scheduler and planning in general.  Individual step and atom time estimates, which
+are of interest not only to Explore but also the scheduler, are available in the execution
+configuration atoms and steps as shown later.  Planned time at this level would summarize setup, 
+acquisition sequence and science sequence time estimates. 
+
+_The following SDL sketches out the general idea but specific details are likely to change._
+
+```
+
+# Individual step time can be divided into categories.  Time spent in each 
+# category will be grouped. 
+type StepTime {
+
+  # Time spent executing configuration changes (e.g., swapping filters)
```

Comment:
... or offsetting the telescope.
I'm just making the point that configuring does not apply to the instrument only.